lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php

```diff
@@ -1551,4 +1565,12 @@ private function generateIdentifierName($identifier)
 
         return strtoupper(dechex(crc32($identifier->getName())));
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isValidIdentifier($name)
+    {
+        return !!preg_match('/^(?:(?:\p{L}|[_@#])(?:\p{L}|\p{Nd}|[$@#_])*|".*"|\[.*\])$/ui', $name);
```

Comment:
Note that `\p{L}` works only if the regex stuff was compiled with enabled unicode: that may or may not be a problem (I don't think it is).

I was more referring to where this reference regex comes from, if it comes from somewhere :-)